package.json

```diff
@@ -25,16 +25,19 @@
     "spec:schemas": "hammer task spec_schemas",
     "test": "npm run spec"
   },
+  "dependencies": {
+    "@coderspirit/nominal-symbols": "^1.0.1"
```

Comment:
I could not avoid adding this dependency given the goals of this PR.

Reply:
I think for type logic like this, the preference would be to implement the types and inference logic inside TypeBox itself rather than to take on remote dependency.